homeassistant/components/sensor/metoffice.py

```diff
@@ -0,0 +1,168 @@
+
+    try:
+        site = datapoint.get_nearest_site(longitude=hass.config.longitude,
+                                          latitude=hass.config.latitude)
+    except dp.exceptions.APIException as err:
+        _LOGGER.error("Received error from Met Office Datapoint: %s", err)
+        return False
+
+    if not site:
+        _LOGGER.error("Unable to get nearest Met Office forecast site")
+        return False
+    else:
+
+        # Get data
+        data = MetOfficeCurrentData(hass, datapoint, site)
+        try:
+            data.update()
+        except (ValueError, dp.exceptions.APIException) as err:
+            _LOGGER.error("Received error from Met Office Datapoint: %s", err)
+            return False
+
+        # Add
+        add_devices([MetOfficeCurrentSensor(site, data, variable)
+                     for variable in config[CONF_MONITORED_CONDITIONS]])
+        return True
+
+
+class MetOfficeCurrentSensor(Entity):
+    """Implementation of a Met Office current sensor."""
+
+    def __init__(self, site, data, condition):
+        """Initialize the sensor."""
+        self.site = site
+        self.data = data
+        self._condition = condition
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return 'Met Office {}'.format(SENSOR_TYPES[self._condition][0])
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self._condition in self.data.data.__dict__.keys():
+            variable = getattr(self.data.data, self._condition)
+            if self._condition == "weather":
+                return variable.text
+            else:
+                return variable.value
+        else:
+            return STATE_UNKNOWN
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return SENSOR_TYPES[self._condition][1]
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        attr = {}
+        attr['Sensor Id'] = self._condition
+        attr['Site Id'] = self.site.id
+        attr['Site Name'] = self.site.name
+        attr['Last Update'] = self.data.lastupdate
+        attr[ATTR_ATTRIBUTION] = CONF_ATTRIBUTION
+        return attr
+
+    def update(self):
+        """Update current conditions."""
+        self.data.update()
+
+
+class MetOfficeCurrentData(object):
+    """Get data from Datapoint."""
+
+    def __init__(self, hass, datapoint, site):
+        """Initialize the data object."""
+        self._hass = hass
+        self._datapoint = datapoint
+        self._site = site
+        self.data = None
+        self.lastupdate = LAST_UPDATE
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Get the latest data from Datapoint."""
+        import datapoint as dp
+        if self.lastupdate != 0 and \
+            ((datetime.datetime.now() - self.lastupdate) <
+             datetime.timedelta(minutes=35)):
```

Comment:
Instead of this, just set `SCAN_INTERVAL` at the top to 35 minutes and also pass that in to Throttle.